### PR TITLE
Add NVIDIA GPUs to pkgci regression tests.

### PR DESCRIPTION
**Pull request by ScottTodd**
Filling in more vendor/device/API coverage.

TODO (later?):
* merge these back into a single workflow [using a matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) once they are stable
* Standardize the set of tests each job runs (include full model tests, which needs Git LFS installed/installable on all runners)
* Rename jobs, runners, etc. for consistency (just using 'gpu' to mean 'NVIDIA T4' is not great)

### CODE REVIEW

File: .github/workflows/pkgci_regression_test_nvidiagpu_cuda.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Installing external TestSuite Python requirements
+        run: |
+          source ${VENV_DIR}/bin/activate
+          python -m pip install -r SHARK-TestSuite/iree_tests/requirements.txt
+      - name: Run external tests - ONNX test suite
+        run: |
+          source ${VENV_DIR}/bin/activate
+          pytest SHARK-TestSuite/iree_tests/onnx/ \
+              -n 4 -rpfE --timeout=30 --retries=2 --durations=20 \
+              --config-files=build_tools/pkgci/external_test_suite/onnx_gpu_cuda_t4.json
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test_nvidiagpu_cuda.yml]: Lovely, this is hanging on some test cases (https://github.com/openxla/iree/actions/runs/8545230629/job/23413351804?pr=16966#step:9:27) , and the pytest timeout didn't catch it. I can try to repro locally.

**Review by saienduri — COMMENTED** [review]
LGTM

**Review by saienduri — APPROVED** [review]

**Comment by ScottTodd** [comment]
I'm a little concerned with how much work updating XFAIL sets will be across CPU, Vulkan, CUDA, and ROCm. https://github.com/openxla/iree/pull/16958 will help a bit, but https://github.com/openxla/iree/pull/16974 still took me ~5 tries to get the lists fully synced for just ROCm x_x

**Comment by saienduri** [comment]
> I'm a little concerned with how much work updating XFAIL sets will be across CPU, Vulkan, CUDA, and ROCm. #16958 will help a bit, but #16974 still took me ~5 tries to get the lists fully synced for just ROCm x_x

Yeah, I was thinking that too, but we will have to maintain separate lists right or maybe we can have a common list that the conftest checks on for tests that fail before it gets to the backend dependent stuff?

**Comment by ScottTodd** [comment]
We could add a common list, but I haven't given much thought to how that would actually work... seems messy if a test starts failing on one backend or newly passing on only some backends.

I think I could script updating the config files from logs (Stella had the start of such a script here: https://gist.github.com/stellaraccident/eb3573840d54a401bca89c456430c255). I've been using a combination of text editor shortcuts and `diff` personally so far. Ideally the test runner itself would generate a config file to "accept" the test results, but I think that would take a bit longer to code into pytest (tried it briefly, it's tricky)

**Pull request merged**